src/QsCompiler/Transformations/CallGraphWalker.cs

```diff
@@ -123,39 +150,51 @@ public override ExpressionKind OnIdentifier(Identifier sym, QsNullable<Immutable
                 {
                     // Type arguments need to be resolved for the whole expression to be accurate
                     // ToDo: this needs adaption if we want to support type specializations
-                    var typeArgs = tArgs;
+                    var typeArgs = QsNullable<ImmutableArray<ResolvedType>>.Null;
 
-                    TypeParamUtils.TryCombineTypeResolutionsForTarget(global.Item, out var typeParamRes, SharedState.typeParameterResolutions.ToArray());
-                    SharedState.typeParameterResolutions = new List<TypeParameterResolutions>();
+                    TypeParamUtils.TryCombineTypeResolutionsForTarget(global.Item, out var typeParamRes, SharedState.TypeParameterResolutions.ToArray());
+                    SharedState.TypeParameterResolutions = new List<TypeParameterResolutions>();
 
-                    if (SharedState.inCall)
+                    if (SharedState.IsInCall)
                     {
                         var kind = QsSpecializationKind.QsBody;
-                        if (SharedState.hasAdjointDependency && SharedState.hasControlledDependency)
+                        if (SharedState.HasAdjointDependency && SharedState.HasControlledDependency)
                         {
                             kind = QsSpecializationKind.QsControlledAdjoint;
                         }
-                        else if (SharedState.hasAdjointDependency)
+                        else if (SharedState.HasAdjointDependency)
                         {
                             kind = QsSpecializationKind.QsAdjoint;
                         }
-                        else if (SharedState.hasControlledDependency)
+                        else if (SharedState.HasControlledDependency)
                         {
                             kind = QsSpecializationKind.QsControlled;
                         }
 
-                        SharedState.graph.AddDependency(SharedState.spec, global.Item, kind, typeArgs, typeParamRes);
+                        SharedState.Graph.AddDependency(
+                            SharedState.CurrentSpecialization,
+                            global.Item, kind, typeArgs,
+                            typeParamRes);
```

Comment:
Do we have a style preference for how to split function arguments across multiple lines? E.g. some style guides require to either have all arguments on one line or split one argument per line.

Reply:
I am not sure if there is a style rule for this. I have just been going by what looks good and consistent to me. If you can find something in the C# style guide, please point it out to me.

Reply:
I don't see anything about this in https://ms-quantum.visualstudio.com/Quantum%20Program/_wiki/wikis/Quantum-OKRs.wiki/487/Coding-Style either. I suppose, it's a personal call then.